lib/bloc/pos_catalog/bloc.dart

```diff
@@ -276,8 +293,130 @@ class PosCatalogBloc with AsyncActionsHandler {
     _loadItems();
   }
 
+  void _loadSelectedReportTimeRange() async {
```

Comment:
All serialization/deserialization to preferences is implemented using a lot of code right now.
Don't you think it would be better to encapsulate it inside PosReportTimeRange?
The standard way we use to serialize models to preferences is toJson/fromJson methods and I think it can be implemented here in _loadSelectedReportTimeRange/_updatePosReportTimeRange methods

Reply:
Changed 🎖️